src/main/java/org/truffleruby/interop/InteropNodes.java

```diff
@@ -947,14 +947,11 @@ public int keyInfoFlagsToBitsNodeZero(boolean existing, boolean readable, boolea
             return 0;
         }
 
+        @TruffleBoundary
```

Comment:
This should not be (doesn't need and make the `throw` deopt every time) a boundary.

Reply:
It saves space in SVM though?

Reply:
Right, a bit.
In that case I'd just remove the specialization since it's impossible, isn't?

Reply:
You'd get an argument error but no helpful information.

Reply:
Which is fine because it's an internal error and shows a bug in the logic of KEY_INFO, no?

Reply:
Maybe we should make this a primitive or `private` so it's clear it's only an internal helper and doesn't expect to be called externally.

Reply:
Since the Ruby logic already tests all those flags, we could keep a single specialization and not even pass `existing` (and let Ruby return 0 no flags are set in the `else` branch).